compose/cli/main.py

```diff
@@ -979,21 +979,23 @@ def up(self, options):
                 raise UserError('--no-start and {} cannot be combined.'.format(excluded))
 
         with up_shutdown_context(self.project, service_names, timeout, detached):
-            try:
-                to_attach = self.project.up(
+            def up(rebuild):
+                return self.project.up(
                     service_names=service_names,
                     start_deps=start_deps,
                     strategy=convergence_strategy_from_opts(options),
-                    do_build=build_action_from_opts(options),
+                    do_build=build_action_from_opts(options) if not rebuild else BuildAction.force,
```

Comment:
This actually doesn't fix the problem. When the build is forced, it still checks the images and throws the NoImageFound exception.

Reply:
Ah, that sucks :(

Can we add a test for it? I know there's a unit test there but it didn't catch anything with my changes, so we probably need something more comprehensive. Once we have a good test we can figure out what the best approach to fixing it is.

Reply:
Alright, I'll work on that.